### ISSUE TITLE
Save statistics in `ValuationResult`

### ISSUE THREAD

**Issue by mdbenito** [aai-institute/pyDVL#288]
- [ ] Statistics like total time, average time between single updates (should be optional)
- [ ] `StoppingCriterion` that updates the status (could be logical or/and)
- [ ] State of the stopping criterion? Difficult to merge in `__add__`
- [ ] Additional info like number of truncations in TMCS ?

**Comment by AnesBenmerzoug** [comment]
I really think the `ValuationResult` class would become too big after this. Perhaps we can split the value part from the rest of it.

**Comment by mdbenito** [comment]
> I really think the `ValuationResult` class would become too big after this. Perhaps we can split the value part from the rest of it. 

Makes sense. I was actually thinking of using a separate statistics object anyway. But if this is not made a property of the valuation result, we are going to have to wrap both into another object, or return multiple objects. Both imply breaking changes to the interfaces...

**Comment by AnesBenmerzoug** [comment]
Perhaps we could consider using a storage backend (a database with tables) similar to how it is done in [Optuna](https://optuna.readthedocs.io/en/stable/) to keep track of values and statistics.

It would decouple many things and we could even fetch results later from the database without having to store them.
They main drawbacks would be that it is one more component that would have to be deployed (unless we somehow make the default local only and use sqlite for that) and that we would have to take into account migrations if we decided to change the schema.

Still I think it's worth considering.

**Comment by mdbenito** [comment]
Hmm... I think we must avoid making a deployment necessary. The joblib
backend is a great addition in this direction, and a local in-memory cache
would also be.

And even if the stats logging is optional, we would be duplicating
functionality in experiment tracking systems. Another thing that we should
avoid IMO. People will typically have made their own choices as to how to
track things, and we need to enable those, not add one on top.

**Comment by mdbenito** [comment]
We can save any information from value updates using the same idea as the `History` stopping criterion. For information coming from the subprocesses we'd have to change the return type of ValueUpdate to include stats, but it seems to me that the history is enough => closing
